src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/VisitSchedulerClient.kt

```diff
@@ -0,0 +1,138 @@
+      .bodyToMono<VisitDto>().block(apiTimeout)
+  }
+
+  fun bookVisitSlot(applicationReference: String): VisitDto? {
+    return webClient.put()
+      .uri("/visits/$applicationReference/book")
+      .accept(MediaType.APPLICATION_JSON)
+      .retrieve()
+      .bodyToMono<VisitDto>().block(apiTimeout)
+  }
+
+  fun changeVisitSlot(applicationReference: String, changeVisitSlotRequestDto: ChangeVisitSlotRequestDto): VisitDto? {
+    return webClient.put()
+      .uri("/visits/$applicationReference/slot/change")
+      .body(BodyInserters.fromValue(changeVisitSlotRequestDto))
+      .accept(MediaType.APPLICATION_JSON)
+      .retrieve()
+      .bodyToMono<VisitDto>().block(apiTimeout)
+  }
+
+  fun changeBookedVisit(reference: String, reserveVisitSlotDto: ReserveVisitSlotDto): VisitDto? {
+    return webClient.put()
+      .uri("/visits/$reference/change")
+      .body(BodyInserters.fromValue(reserveVisitSlotDto))
+      .accept(MediaType.APPLICATION_JSON)
+      .retrieve()
+      .bodyToMono<VisitDto>().block(apiTimeout)
+  }
+
+  fun cancelVisit(visitCancelDto: VisitCancelDto): VisitDto? {
+    return webClient.put()
+      .uri("/visits/${visitCancelDto.reference}/cancel")
+      .body(BodyInserters.fromValue(visitCancelDto.outcome))
+      .accept(MediaType.APPLICATION_JSON)
+      .retrieve()
+      .bodyToMono<VisitDto>().block(apiTimeout)
+  }
+
+  fun getVisitSupport(): List<SupportTypeDto>? {
+    return webClient.get()
+      .uri("/visit-support")
+      .accept(MediaType.APPLICATION_JSON)
+      .retrieve()
+      .bodyToMono<List<SupportTypeDto>>().block(apiTimeout)
+  }
+
+  fun getVisitSessions(prisonId: String, prisonerId: String?, min: Long?, max: Long?): List<VisitSessionDto>? {
+    return webClient.get()
+      .uri("/visit-sessions") {
+        visitSessionsUriBuilder(prisonId, prisonerId, min, max, it).build()
+      }
+      .accept(MediaType.APPLICATION_JSON)
+      .retrieve()
+      .bodyToMono<List<VisitSessionDto>>().block(apiTimeout)
+  }
+
+  fun getSupportedPrisons(): List<String>? {
+    return webClient.get()
+      .uri("/config/prisons/supported")
+      .accept(MediaType.APPLICATION_JSON)
+      .retrieve()
+      .bodyToMono<List<String>>().block(apiTimeout)
+  }
+
+  private fun visitSearchUriBuilder(visitSearchRequestFilter: VisitSearchRequestFilter, uriBuilder: UriBuilder): UriBuilder {
+    uriBuilder.queryParamIfPresent("prisonId", Optional.ofNullable(visitSearchRequestFilter.prisonCode))
+    uriBuilder.queryParamIfPresent("prisonerId", Optional.ofNullable(visitSearchRequestFilter.prisonerId))
+    uriBuilder.queryParam("visitStatus", visitSearchRequestFilter.visitStatusList)
+    uriBuilder.queryParamIfPresent("startDateTime", Optional.ofNullable(visitSearchRequestFilter.startDateTime))
+    uriBuilder.queryParamIfPresent("endDateTime", Optional.ofNullable(visitSearchRequestFilter.endDateTime))
+    uriBuilder.queryParamIfPresent("visitorId", Optional.ofNullable(visitSearchRequestFilter.visitorId))
```

Comment:
if visitorId is null I assume that the parameter is not added and it dose not equal &visitorId=null?

Reply:
Yes thats right. Only adds it if not null.